packages/interactivity-router/src/index.js

```diff
@@ -117,21 +126,45 @@ export const { state, actions } = store( 'core/router', {
 				setTimeout( resolve, options.timeout ?? 10000 )
 			);
 
+			// Don't update the navigation status immediately, wait 400 ms.
+			const timeout = setTimeout( () => {
+				if ( navigatingTo === href ) {
+					navigation.hasStarted = true;
+					navigation.hasFinished = false;
+					navigation.message = navigation.texts.loading;
+				}
+			}, 400 );
+
 			const page = yield Promise.race( [
 				pages.get( url ),
 				timeoutPromise,
 			] );
 
+			// Dismiss loading message if it hasn't been added yet.
+			clearTimeout( timeout );
+
 			// Once the page is fetched, the destination URL could have changed
 			// (e.g., by clicking another link in the meantime). If so, bail
 			// out, and let the newer execution to update the HTML.
 			if ( navigatingTo !== href ) return;
 
+			navigation.hasStarted = false;
+			navigation.hasFinished = true;
+			// Announce that the page has been loaded. If the message is the
+			// same, we use a no-break space similar to the @wordpress/a11y
+			// package: https://github.com/WordPress/gutenberg/blob/c395242b8e6ee20f8b06c199e4fc2920d7018af1/packages/a11y/src/filter-message.js#L20-L26
+			navigation.message =
+				navigation.texts.loaded +
+				( navigation.message === navigation.texts.loaded
+					? '\u00A0'
+					: '' );
```

Comment:
Shouldn't we skip this when there was a timeout? I'm not sure the screen reader will speak before the full page reload happens.

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/58377/commits/6ab84a11f44a8341558e531620990bdb86545ccc.